# Data Analyst

#### Summary: 
B.Sc. in Computer Science with research and internship experience in artificial intelligence, data science, and software development. Skilled in developing machine learning models, analyzing data, and creating interactive applications. Contributed to projects like breast cancer detection and climate data visualization. Interested in roles involving artificial intelligence, predictive analytics, or software development, with a long-term goal of contributing to innovative research and technology-driven solutions.

#### Technical Skills:

- Programming Languages: _Proficient in Python, SQL, C++ & JavaScript; experience with HTML, CSS, and React.js._
- Data Science and Analytics: _statistical analysis, machine learning, data visualization, and predictive modeling._
- Tools and Development: _Proficient in Power BI, Excel, Git & version control systems; experienced with Python libraries like Matplotlib, Plotly, and Seaborn for data visualization; adept in front-end development using React.js and working with tools like VS Code / PyCharm, and LaTeX._
  
## üéì Education    		
- B.Sc., Computer Science | <a href="https://www.hitecuni.edu.pk/" target="_blank">HITEC University Taxila</a> (_Sep 2024_)

## üî¨ Research Experience
**Research Contributor @<a href="https://www.hitecuni.edu.pk" target="_blank">HITEC</a> (_2022 - 2023_)**
- Collaborated on a breast cancer detection project using machine learning techniques, contributing to advancements in healthcare technology.
- Spearheaded **data preprocessing** and **feature engineering** to improve model accuracy.
- Assisted in the development and evaluation of predictive models for breast cancer detection.
- Analyzed model performance and provided actionable insights to refine predictive capabilities.
  
**Independent Researcher @ <a href="https://zenodo.org/records/13900889" target="_blank">A 270-Year Overview of Global Temperature Variations</a> (_2024_)**
- Conducted an extensive analysis of global temperature trends spanning from 1743 to 2013, focusing on long-term climatic shifts and their potential impact on future climate policy.
- Utilized advanced **data visualization** and **statistical techniques** to identify significant temperature fluctuations across global regions and highlight their implications for climate resilience.
- Developed interactive visualizations to present findings, facilitating a deeper understanding of climate change patterns and their influence on global warming mitigation strategies.
- Contributed to the field of climate science by emphasizing the importance of **data-driven decision-making** in addressing the challenges of climate adaptation.

## üíº Work experience 
**Data Science Intern @ <a href="https://coderscave.in/" target="_blank">Coderscave </a>, India (_Decembe 2020 - Present_)**
- Gained hands-on experience in data analysis, statistical modeling, and machine learning using Python and SQL, working with real-world datasets.
- Developed an **Email Spam Filter** using **Natural Language Processing (NLP)** and machine learning algorithms, improving email classification accuracy. 
- Conducted **time series analysis** on stock price data to identify trends and forecast future market behavior, demonstrating advanced analytical skills.

## üíª Programming and data analysis skills
- Proficient in **Python**, **SQL**, **MATLAB**, and JavaScript for data analysis, statistical modeling, and machine learning tasks.
- Experienced in developing **machine learning models**, including natural language processing (NLP) applications like Email Spam Filters and time series forecasting for stock prices.
- Skilled in **data visualization** using Python libraries such as Matplotlib, Seaborn, and Plotly to create interactive visualizations for large-scale datasets.
- Expertise in **statistical analysis** and predictive modeling for analyzing complex datasets, including applications in climate data analysis and business data insights.
- Familiar with **data wrangling** and **feature engineering** techniques to process and clean datasets for machine learning model training.

**Additional Skills:**
  
- Git/GitHub for version control and collaborative coding.
- Strong foundation in front-end development using React.js, HTML, and CSS.
- Hands-on experience with Power BI and Excel for data visualization and analysis.
- Skilled in data cleaning and transformation for machine learning using Python and SQL.

## üå± R Packages developed
### ‚úÖ PBGeno
<a href="https://github.com/Allan-gitrepos/pbgeno" target="_blank">GitHub</a>

Developed pbgeno, an R package to streamline data analysis workflows for plant breeders. The package provides functions for calculating genetic distances, clustering genotypes, estimating diversity statistics, creating publication-quality visualizations, and automating routine tasks. Key features include calculating genetic distance matrices, structure-based clustering, polymorphism quantification, and converting proprietary marker genotypes into standardized formats for genome-wide association mapping. 

### ‚úÖ PBPerfect
**PBPerfect** <a href="https://allanbiotools.shinyapps.io/pbperfect/" target="_blank">(Visit Page)</a> is a interactive web tool enabling reproducible multivariate analysis with visualization of phenotypic and genotypic data. It features basic statistics, experimental designs, SSR workflows, multivariate analysis, mating designs, and dynamic graphics with outputs exported as publication-standard tables and graphics requiring no further formatting.

### ‚úÖ PBMLT: Plant Breeding Multilocation Trail Data Analysis Software

**PBMLT**<a href="https://allanbiotools1.shinyapps.io/PBMLT/" target="_blank">(Visit Page)</a> is a comprehensive and user-friendly platform that provides plant breeders with an all-in-one solution for analyzing multi-environment trial data through:

- Powerful Analysis of Variance (ANOVA) to explore significance of variation across locations and treatments.
- Additive Main Effects and Multiplicative Interaction (AMMI) analysis for in-depth genotype-environment interaction studies.
- Calculation of essential AMMI-based stability indices like ASV, ASTAB, ASI, MASI, SIPC, ZA for identifying adaptable lines.
- Evaluation of overall productivity using metrics such as mean AVAMGE for high-yielding genotype selection.
- Scaled stability measures like SSIASTAB and ASI_SSI for ranking lines based on trait stability.
- Interactive visualizations including Biplots, GGE plots, WASS plots for straightforward interpretation of complex data.
- Centralized platform integrating meta-analysis, statistical analytics, and genotype-environment interaction analysis.

### ‚úÖ PBlinkagemap

**PBlinkagemap** <a href="https://allanbiotools1.shinyapps.io/PBLinkagemap/" target="_blank">(Visit Page)</a> enables easy creation of linkage maps and identification of associated quantitative trait loci (QTLs) from genomic and phenotypic datasets. 

It allows users to:
- Import chromosome, marker, map distance and trait score data
- Interactively explore results on linkage maps
- Visualize QTL locations and effects
By handling computationally intensive linkage analysis and mapping behind the scenes, PBlinkagemap makes it simple for users to go from datasets to QTL discovery through an intuitive interface.

### ‚úÖ PB-GWAS üß¨

**PB-GWAS** <a href="https://allanbiotools1.shinyapps.io/PBgwas/" target="_blank">(Visit Page)</a> makes powerful genome-wide association studies accessible through an easy-to-use web app üë©‚Äçüíª

Key features:
üì• File upload in 4 clicks  
üèÉ‚Äç‚ôÇÔ∏è One-click GWAS launch  
‚öôÔ∏è Adjust parameters via sidebar   
üìà Interactive result plotting  
üìÑ Full PDF report downloading

By eliminating coding barriers, PB-GWAS allows both new and advanced users to leverage GAPIT workflows with no programming expertise required!. Whether you want to map simple or complex traits, PB-GWAS provides the automated analysis to accelerate discoveries üî¨

### ‚úÖ PBHaploMineR üß¨

**PBHaploMineR** <a href="https://github.com/Allan-gitrepos/PBhaplominerR" target="_blank">(Visit Page)</a> provides a toolkit to streamline pangenome haplotype mining and comparison from next-generation sequencing data. This R package aims to make large-scale haplotype analysis efficient and accessible for species with reference pangenomes.

**Key Features:**
* **Sequence Import** - Functions to import raw reads from multiple platforms and store in standardized schema
* **Haplotype Calling** - Optimized algorithms for pangenome-wide haplotype calling, incorporating structural variation
* **HapViz** - Interactive visualization system to explore and compare haplotypes in context of pangenome structure
* **HapCompare** - Statistically compare haplotypes between groups of samples/accessions and identify associated genomic signatures
* **Parallelization** - Built-in parallelization to scale analyses across HPC infrastructure

PBHaploMineR is still under development and testing. ETA for first stable version is Q1 2024.

## üé§ Workshop and Conferences
- Attended the workshop on "<a href="https://twitter.com/GenebankICRISAT/status/1169137453472669696" target="_blank">A15 DArTSeq Data Analysis</a>" at <a href="https://www.cimmyt.org/" target="_blank">CIMMYT</a>, Mexico
- Attended the international conference on <a href="https://twitter.com/GenebankICRISAT/status/1158297753371308032" target="_blank">"Neglected and Underutilized Crop Species for Food, Nutrition, Energy and Environment"</a> at NIPGR, New Delhi. Awarded a travel grant in recognition of contribution. 
- Attended the workshop on **<a href="https://www.facebook.com/media/set/?vanity=natesan.senthil&set=a.5804435956243670" target="_blank">HPLC: "Principles and Applications in Plant Metabolomics"</a>**" at TNAU,Coimbaotore.
- Attended the workshop on **<a href="https://www.covaichronicle.com/english/contentview/natu?fbclid=IwAR0fA8w7ubx__cGaKnmE-68-tmRLg92XHs1Rf-HnhQzRjmFc_calVar8yFo" target="_blank">Molecular Modelling and Docking</a>** at TNAU, Coimbatore.
- Presented poster titled "**Genes for salt tolerance revealed by functional metagenomics in rice**" at 6<sup>th</sup> National conference AC&RI, TNAU, Trichy.

## ‚úçÔ∏è Articles & Blogs
- <a href="https://fayzankj.medium.com/" target="_blank">Medium Articles</a>

## üìú Publications

1. Allan Victor, Mani Vetriventhan, Ramachandran Senthil, S. Geetha, Santosh Deshpande, Abhishek Rathore, Vinod Kumar, Prabhat Singh, Surender Reddymalla, and V√¢nia CR Azevedo. **"Genome-wide DArTSeq genotyping and phenotypic based assessment of within and among accessions diversity and effective sample size in the diverse sorghum, pearl millet, and pigeonpea landraces."**_Frontiers in Plant Science_ 11 (2020): 587426. (DOI: <a href="https://doi.org/10.3389/fpls.2020.587426" target="_blank">https://doi.org/10.3389/fpls.2020.587426</a>)
2. Allan, V. (2023) **‚ÄòPB-Perfect: A Comprehensive R-Based Tool for Plant Breeding Data Analysis‚Äô**, PB - Perfect. Available at: <a href="https://allanbiotools.shinyapps.io/pbperfect/" target="_blank">https://allanbiotools.shinyapps.io/pbperfect/</a>.
3. Backiyalakshmi, C., Mani Vetriventhan, Santosh Deshpande, C. Babu, Allan Victor, D. Naresh, Rajeev Gupta, and Vania CR Azevedo. **"Genome-wide assessment of population structure and genetic diversity of the global finger millet germplasm panel conserved at the ICRISAT Genebank."** _Frontiers in Plant Science_ 12 (2021): 692463. (DOI: <a href="https://doi.org/10.3389/fpls.2021.692463" target="_blank">https://doi.org/10.3389/fpls.2021.692463</a>)
4. Vetriventhan, Mani, Hari D. Upadhyaya, Vania CR Azevedo, Allan Victor, and Seetha Anitha. **"Variability and trait‚Äêspecific accessions for grain yield and nutritional traits in germplasm of little millet (_Panicum sumatrense_ Roth. Ex. Roem. & Schult.)."** _Crop Science_ 61, no. 4 (2021): 2658-2679. (DOI: <a href="https://doi.org/10.1002/csc2.20527" target="_blank">https://doi.org/10.1002/csc2.20527</a>)
5. Jagadesh, M., Duraisamy Selvi, Subramanium Thiyageshwari, Thangavel Kalaiselvi, Allan Victor, Munmun Dash, Keisar Lourdusamy, Ramalingam Kumaraperumal, Pushpanathan Raja, and U. Surendran. **"Exploration of microbial signature and carbon footprints of the Nilgiri Hill Region in the Western Ghats global biodiversity hotspot of India."** _Applied Soil Ecology_ (2023): 105176 (DOI: <a href="https://doi.org/10.1016/j.apsoil.2023.105176" target="_blank">https://doi.org/10.1016/j.apsoil.2023.105176</a>).
6. Jagadesh, M., Cherukumalli Srinivasarao, Duraisamy Selvi, Subramanium Thiyageshwari, Thangavel Kalaiselvi, Aradhna Kumari, Santhosh Kumar Singh, Allan Victor **"Quantifying the Unvoiced Carbon Pools of the Nilgiri Hill Region in the Western Ghats Global Biodiversity Hotspot‚ÄîFirst Report."** _Sustainability_ 15, no. 6 (2023): 5520. (DOI: <a href="https://doi.org/10.3390/su15065520" target="_blank">https://doi.org/10.3390/su15065520</a>)
7. Jagadesh, M., Duraisamy Selvi, Subramanium Thiyageshwari, Cherukumalli Srinivasarao, Thangavel Kalaiselvi, Keisar Lourdusamy, Ramalingam Kumaraperumal, and Victor Allan. **"Soil Carbon Dynamics Under Different Ecosystems of Ooty Region in the Western Ghats Biodiversity Hotspot of India."** _Journal of Soil Science and Plant Nutrition_ 23, no. 1 (2023): 1374-1385. (DOI: <a href="https://doi.org/10.1007/s42729-023-01129-2" target="_blank">https://doi.org/10.1007/s42729-023-01129-2</a>)
8. Allan Victor., N. Meenakshi Ganesan, R. Saraswathi, R. Gnanam, and C. N. Chandrasekhar. **"Exploring the phenotypic diversity of rice: A multivariate analysis of local landraces and elite cultivars of Tamil Nadu and Exotic Lines."** _Electronic Journal of Plant Breeding_ 14, no. 3 (2023): 857-866. (DOI: <a href="https://ejplantbreeding.org/index.php/EJPB/article/view/4804" target="_blank">10.37992/2023.1403.099</a>)
9. Allan Victor, S. Geetha, Mani Vetriventhan, and V√¢nia CR Azevedo. **"Genetic diversity analysis of geographically diverse landraces and wild accessions in sorghum."** _Electronic Journal of Plant Breeding_ 11, no. 03 (2020): 760-764. (DOI: <a href="https://www.ejplantbreeding.org/index.php/EJPB/article/view/3575" target="_blank">https://doi.org/10.37992/2020.1103.125</a>)
    

## üìö References

|     |     |
|:----|:----|
| **Name:** | **Dr. Vania de Azevedo** |
| **Position:** | Former Head, Plant Genetic Resources |
| **Organization:** | ICRISAT, Hyderabad, India |
| **E-mail:** | <a href="mailto:azevedovcr@gmail.com">azevedovcr@gmail.com</a> |
| **LinkedIn:** | <a href="https://www.linkedin.com/in/v%C3%A2nia-c-renn%C3%B3-azevedo-21493449/" target="_blank">Visit Page</a> |

|     |     |
|:----|:----|
| **Name:** | **Dr. Mani Vetriventhan** |
| **Position:** | Senior Scientist, Plant Genetic Resources |
| **Organization:** | ICRISAT, Hyderabad, India |
| **E-mail:** | <a href="mailto:M.Vetriventhan@cgiar.org">M.Vetriventhan@cgiar.org</a> |
| **LinkedIn:** | <a href="https://www.linkedin.com/in/vetriventhan-mani-26468322/" target="_blank">Visit Page</a> |

|     |     |
|:----|:----|
| **Name:** | **Mr. Rajaguru Bohar** |
| **Position:** | Regional Genotyping Coordinator (South Asia) / Senior Scientist (Project management) |
| **Organization:** | CIMMYT |
| **E-mail:** | <a href="mailto:wishmeguru@gmail.com">wishmeguru@gmail.com</a> |
| **LinkedIn:** | <a href="https://www.linkedin.com/in/rajaguru-bohar-5038762a/" target="_blank">Visit Page</a> |
|     |     |


## üìû Contact

| **Name**   | Allan. V | 
| **E-mail**   | <a href="mailto:albertoogy@gmail.com">albertoogy@gmail.com</a> | 
| **LinkedIn**   | <a href="https://www.linkedin.com/in/allan-victor-6919b4152/" target="_blank">Visit Page</a> | 
